### PR TITLE
perf(ui): Lazy load react-date-range

### PR DESCRIPTION
**Pull request by scttcper**
Lazy load react-date-range as it is a large package. Displays a spinner until it has loaded.

Videos recorded with network slow 3g enabled

https://user-images.githubusercontent.com/1400464/111241195-e3123f80-85b9-11eb-940a-0fa5cf449d56.mp4

https://user-images.githubusercontent.com/1400464/111241213-e9082080-85b9-11eb-8f49-ed3cc7de58bd.mp4

### CODE REVIEW

File: src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateRange/dateRangeWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+  .rdrDay {
+    height: 2.5em;
+  }
+
+  .rdrMonthPicker select,
+  .rdrYearPicker select {
+    background: none;
+    font-weight: normal;
+    font-size: 16px;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateRange/dateRangeWrapper.tsx:110]:
```suggestion
    font-size: ${p => p.theme.fontSizeLarge};
```

File: src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateRange/index.tsx [hunk 2]
```diff
@@ -197,19 +200,27 @@ class DateRange extends React.Component<Props, State> {
...
-        />
+        <React.Suspense
+          fallback={
+            <Placeholder width="342px" height="254px">
+              <LoadingIndicator />
+            </Placeholder>
+          }
+        >
+          <DateRangePicker
+            rangeColors={[theme.purple300]}
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateRange/index.tsx:211]: should we wrap DateRange with `withTheme`?

File: src/sentry/static/sentry/app/views/settings/components/forms/calendarWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import 'react-date-range/dist/styles.css';
+import 'react-date-range/dist/theme/default.css';
+
+export {Calendar as default} from 'react-date-range';
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/components/forms/calendarWrapper.tsx:4]: should we maybe name this file `reactDateRangeCalendar` or `Calendar` or `CalendarField` ?

File: src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateRange/index.tsx [hunk 4]
```diff
@@ -197,19 +200,27 @@ class DateRange extends React.Component<Props, State> {
...
-              startDate: moment(start).local().toDate(),
-              endDate: moment(end).local().toDate(),
-              key: 'selection',
-            },
-          ]}
-          minDate={minDate}
-          maxDate={maxDate}
-          onChange={this.handleSelectDateRange}
-        />
+        <React.Suspense
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/organizations/timeRangeSelector/dateRange/index.tsx:203]: Would be nice if we could get this module loaded when the date range is first opened. That might save some spinner time, as the module could be loaded while the user is moving down to the 'absolute date' option.

File: src/sentry/static/sentry/app/views/settings/components/forms/datePickerField.tsx [hunk 5]
```diff
@@ -28,6 +27,10 @@ function handleChangeDate(
   close();
 }
 
+const Calendar = React.lazy(
+  () => import(/* webpackChunkName: "Calendar" */ './calendarWrapper')
+);
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/components/forms/datePickerField.tsx:32]: Is there anywhere in the application that we're using the datePickerField? I can't seem to find one.

Comment (scttcper) [reply src/sentry/static/sentry/app/views/settings/components/forms/datePickerField.tsx:32]: we are not using it right now, it has a storybook though

Comment (markstory) [reply src/sentry/static/sentry/app/views/settings/components/forms/datePickerField.tsx:32]: We should probably send it to the trash instead :smile:

**Review by markstory — APPROVED** [review]
Awesome stuff. Nice to see vendor.js down by 6%

**Comment by benvinegar** [comment]
Good find!

**Pull request merged**